### PR TITLE
Update UniPC to support 1D diffusion.

### LINKED ISSUE #5198: Add 1D diffusion support to UniPC scheduler
UniPC has been impressive in generating diffusion audio for both singing and speaking. However, the current UniPC implementation in the diffusers repo only supports 2D dimensions. Adding 1D support to the UniPC scheduler would be a great enhancement.

Example projects for reference:
- [DiffSinger](https://github.com/openvpi/DiffSinger)
- [Fish Diffusion](https://github.com/fishaudio/fish-diffusion)
- [Diffusion-SVC](https://github.com/CNChTu/Diffusion-SVC)

### PR DESCRIPTION
**Pull request by leng-yue**
# What does this PR do?

Fixes #5198 by updating only the einsum dimensions. This change should also add support for 3D diffusion.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/diffusers/blob/main/CONTRIBUTING.md)?
- [ ] Did you read our [philosophy doc](https://github.com/huggingface/diffusers/blob/main/PHILOSOPHY.md) (important for complex PRs)?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/diffusers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [x] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Comment by leng-yue** [comment]
Fixed, seems these errors are caused by the latest commit.

File: src/diffusers/utils/testing_utils.py [hunk 1]
```diff
@@ -69,7 +69,8 @@
             ) from e
         logger.info(f"torch_device overrode to {torch_device}")
     else:
-        torch_device = "cuda" if torch.cuda.is_available() else "cpu"
+        # torch_device = "cuda" if torch.cuda.is_available() else "cpu"
+        torch_device = "cpu"
```
Comment (patrickvonplaten) [inline src/diffusers/utils/testing_utils.py:73]: We can't do this change I'm afraid

Comment (leng-yue) [reply src/diffusers/utils/testing_utils.py]: This was updated by mistake, fixed.

**Comment by patrickvonplaten** [comment]
Failing tests are unrelated - merging

**Comment by patrickvonplaten** [comment]
Good job @leng-yue !

**Pull request merged**